src/iter/collect/mod.rs

```diff
@@ -88,55 +89,56 @@ impl<'c, T: Send + 'c> Collect<'c, T> {
     where
         F: FnOnce(CollectConsumer<'_, T>) -> CollectResult<'_, T>,
```

Comment:
I think the closure signature here is why the lifetime parameter wants to be invariant. Since it's internal apis, it's our choice if we want to keep that detail. The invariance just means that you can't play tricks and substitute some in other return value.

Reply:
Thanks, that is a nice guarantee, and zero-cost since it's a function of borrowck. I've added a comment so we can remember this reason later. :slightly_smiling_face: